### ISSUE TITLE
Vav Holam Shifts

### ISSUE THREAD

**Issue by AsherHandaly** [charlesLoder/havarot#21]
Hello Charles, 

Thank you so much for your continued work on this fantastic library. I'm experiencing an issue with a Vav Holam shifting to the previous letter after initializing a Text object. Any suggestions on what may be going wrong? I'm using version 0.7.2 with Node for reference.

Word passed to Text() = א֑וֹר
Syllable returned = אֹ֑ור

Thank you!!

**Comment by charlesLoder** [comment]
@AsherHandaly I'm just happy to know someone else is using it!

I remember all the intricacies around this being confusing to me so I'm going to try to write out the greater context of what's happening. It's not a bug, but it is different than how most texts encode Hebrew.

### Terms

|Term|Defintion|
|:---:|:---:|
| HOLAM | the unicode char U+05B9 |
| _holam male_ | an orthographic convention where a VAV is used in conjunction with a holam |
| _holam haser_ | an orthographic convention where only a holam is used |

### Encoding

There is variation with how to encode the _holam male_:

1. the first is encode it like your example above — C + VAV + HOLAM ( where C = any consonant)
2. the second is how `havarot` standardizes it — C + HOLAM + VAV

### Problem with the first encoding

The [proposal for HOLAM HASER FOR VAV](https://www.unicode.org/L2/L2004/04310-n2840-holam-haser.pdf) (see section D) gives lots of good explanations. 

The gist of the problem is that the sequence of VAV + HOLAM is potentially ambiguous. Is the HOLAM meant to be read as the vowel of the VAV (i.e. [vo])? Or, is it mean to be read as a _holam male_ construction (e.g. just as [o])?

You cannot tell to which consonant the HOLAM is bound solely by its position in the text.

### A potential solution

In order to resolve that ambiguity, the HOLAM HASER FOR VAV code point (U+05BA) was introduced.

According to this solution: 

- in order to represent a regular _holam haser_ construction, you just encode: C + HOLAM
- in order to represent a _holam male_ construction, you encode: C + VAV + HOLAM
- in order to represent a _holam haser_ construction where the C is a VAV, you encode: VAV +  HOLAM HASER FOR VAV.

The  HOLAM HASER FOR VAV point prints differently on the VAV too (see p.4 of the proposal).

I, personally, find this a bit needlessly complex. It may work well for _printing_, but in terms of linguistics, I went with a different approach.

### Havarot approach

The basis of my approach is to keep the consonant and vowel of a Cluster next to each other. Because the vowel is to be read in conjunction with the consonant, it makes sense to keep those two together in the encoding as well.

To put it another way — in "אוֹר" the _holam_ is meant to be read on the aleph (i.e. ['or]) not on the vav (i.e. ['ovr]).

Then, instead of resorting to using an entirely different character to represent a _holam haser_ on a VAV like in the above solution, just encode the HOLAM right after the VAV. 

In other words, instead of encoding "a _holam haser_ construction where the C is a VAV" as VAV +  HOLAM HASER FOR VAV, just use VAV + HOLAM (e.g. עָוֹן).

Another way to think about it is that the HOLAM is bound to a consonant according to its position.

---

I hope that's clear! And would love to know if there are any pitfalls to my method
